src/Sarif.Converters/ClangTidyConverter.cs

```diff
@@ -0,0 +1,171 @@
+
+        internal static Result CreateResult(ClangTidyDiagnostic entry)
+        {
+            entry = entry ?? throw new ArgumentNullException(nameof(entry));
+
+            Result result = new Result()
+            {
+                RuleId = entry.DiagnosticName,
+                Message = new Message { Text = entry.DiagnosticMessage.Message },
+                Kind = ResultKind.Fail
+            };
+
+            // no level infomation in Clang-Tidy report
+            result.Level = FailureLevel.Warning;
+
+            Region region = new Region()
+            {
+                CharOffset = entry.DiagnosticMessage.FileOffset,
+            };
+
+            Uri analysisTargetUri = new Uri(entry.DiagnosticMessage.FilePath, UriKind.RelativeOrAbsolute);
+
+            var physicalLocation = new PhysicalLocation
+            {
+                ArtifactLocation = new ArtifactLocation
+                {
+                    Uri = analysisTargetUri
+                },
+                Region = region
+            };
+
+            Location location = new Location()
+            {
+                PhysicalLocation = physicalLocation
+            };
+
+            result.Locations = new List<Location>()
+            {
+                location
+            };
+
+            if (entry.DiagnosticMessage.Replacements.Count > 0)
+            {
+                IList<Replacement> replacements = new List<Replacement>();
+
+                foreach (ClangTidyReplacement fix in entry.DiagnosticMessage.Replacements)
+                {
+                    Replacement replacement = new Replacement();
+
+                    replacement.DeletedRegion = new Region
+                    {
+                        CharLength = fix.Length,
+                        CharOffset = fix.Offset,
+                    };
+
+                    if (!string.IsNullOrEmpty(fix.ReplacementText))
+                    {
+                        replacement.InsertedContent = new ArtifactContent
+                        {
+                            Text = fix.ReplacementText
+                        };
+                    }
+
+                    replacements.Add(replacement);
+                }
+
+                var sarifFileChange = new ArtifactChange
+                {
+                    ArtifactLocation = new ArtifactLocation
+                    {
+                        Uri = analysisTargetUri
+                    },
+                    Replacements = replacements
+                };
+
+                Fix sarifFix = new Fix(description: null, artifactChanges: new List<ArtifactChange>() { sarifFileChange }, properties: null);
+                result.Fixes = new List<Fix> { sarifFix };
+            }
+
+            return result;
+        }
+
+        private static (List<ReportingDescriptor>, List<Result>) ExtractRulesAndResults(ClangTidyLog log)
+        {
+            var rules = new List<ReportingDescriptor>();
+            var ruleIds = new HashSet<string>();
```

Comment:
u could replace by dict<string, reportingdescriptor>

Reply:
add the stringcomparison to ignore case

Reply:
sounds good, fixed